Add network recommended max fee option

danielgranhao opened the pull request:
Closes #460 

TODO:

- [x] Docs

dangeross commented on crates/breez-sdk/core/src/models/mod.rs:
```diff
@@ -449,22 +449,47 @@ impl Config {
     Fixed { amount: u64 },
     // Relative fee rate in satoshis per vbyte
     Rate { sat_per_vbyte: u64 },
+    // Current network recommended fee with a leeway in satoshis per vbyte
```
Feels like it needs more explanation that it's calculated at request/claim time

danielgranhao commented:
@roeierez @dangeross Made a new change: reverted the `max_fee` type in the claim error to `Fee` because I noticed while writing docs that providing the variant `NetworkRecommended` when a claim fails isn't very helpful. Now, if the new option is selected, the error contains the fee rate that ends up being used as max, which is much more helpful, as it can be compared to the required fee rate.

Docs were also added.

dangeross reviewed (approved):
LGTM
